igrill.py

```diff
@@ -177,8 +193,13 @@ def run(self):
                 while True:
                     temperature = device.read_temperature()
                     battery = device.read_battery()
-                    utils.publish(temperature, battery, self.mqtt_client, self.topic, device.name)
-                    logging.debug("Published temp: {} and battery: {} to topic {}/{}".format(temperature, battery, self.topic, device.name))
+                    heatingelement = device.read_heating_elements()
```

Comment:
If you have done what I suggest in other places, heatingelement should now be None if the device is anything other than a pulse_2000

Reply:
Nitpick: heating_element is a better name.